GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace GenericsHomework;
+
+public class Node<TVal> where TVal : notnull
+{
+    public Node(TVal val)
+    {
+        Val = val ?? throw new ArgumentNullException($"Node<{typeof(TVal)}> value cannot be null");
+        Next = this;
+    }
+
+    public TVal Val
+    {
+        get;
+        set;
+    }
+
+    public Node<TVal> Next
+    {
+        get;
+        private set;
+    }
+
+    public void Append(TVal node)
+    {
+        if(Exists(node))
+        {
+            throw new ArgumentException("This value already exists. Value must be unique.");
+        }
+
+        Node<TVal> node1 = new(node);
```

Comment:
```suggestion
        Node<TVal> newNode = new(node);
```
maybe a better name, adding the number 1 after node is not very clear